src/public/ui.h

```diff
@@ -29,6 +41,12 @@ const char* Warning();
 /// Returns a Tool, or NULL if Ninja should exit.
 const Tool* ChooseTool(const std::string& tool_name);
 
+/// Execute ninja as the main ninja binary would
+/// Does not return, prefering to exit() directly
+/// to avoid potentially expensive cleanup when  destructuring
+/// Ninja's state.
+NORETURN void Execute(int argc, char** argv);
```

Comment:
Of course this is already the existing behavior, but I'd wager that the vast majority of potential library users aren't going to be amused if a shared library that they load makes their application terminate.